homeassistant/components/samsungtv/remote.py

```diff
@@ -49,3 +50,14 @@ async def async_send_command(self, command: Iterable[str], **kwargs: Any) -> Non
 
         for _ in range(num_repeats):
             await self._bridge.async_send_keys(command_list)
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn the remote on."""
+        if self._turn_on_action:
+            await self._turn_on_action.async_run(self.hass, self._context)
+        elif self._mac:
+            await self.hass.async_add_executor_job(self._wake_on_lan)
+        else:
+            raise HomeAssistantError(
+                f"Entity {self.entity_id} does not support this service."
+            )
```

Comment:
I wasn't sure if we should implement `wake_on_lan` on the remote entity, like it is on the `media_player` entity, or if we should encourage people to setup the device trigger.
cc @bdraco / @MartinHjelmare as you had an opinion on the previous discussions

Reply:
Personally, I'd like automations to become better at allowing users to hook up a trigger where the source integration device has a mac and host address with the wake on lan service. I don't think every integration should be responsible for that, as it doesn't need to be.

Eg, if no action is registered and the trigger fires, ask the user if they want to configure the wake on lan service as action.

Reply:
That is a good goal... but I have no idea where to start and TBH I don't have the motivation for it right now.

For now I duplicated the functionality from the media player, and wondering if it is an OK stop-gap until the functionality you describe is available/is documented/has a sample....

Reply:
It would be a long term thing.

Thd current code is ok.

Reply:
Thanks for confirming 👍